objects/src/accounts/account_id.rs

```diff
@@ -240,6 +241,42 @@ impl TryFrom<u64> for AccountId {
     }
 }
 
+impl TryFrom<&str> for AccountId {
+    type Error = AccountError;
+
+    /// Expects the string to start with `0x`.
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        miden_crypto::utils::hex_to_bytes(value)
+            .map_err(|err| AccountError::HexParseError(err.to_string()))
+            .and_then(|mut bytes: [u8; 8]| {
+                // `bytes` ends up being parsed as felt, and the input to that is assumed to be little-endian
+                // so we need to reverse the order
+                bytes.reverse();
```

Comment:
Are we sure this will not cause confusion in the future? It feels like because we are starting from a String type, big-endianness will be assumed. Furthermore, the [docs explain how the information is encoded within an Account Id](https://0xpolygonmiden.github.io/miden-base/architecture/accounts.html#account-id), and the disclaimer says that Miden uses little-endianness as default but big-endianness for readability which leaves some room for ambiguity. 

I don't have a strong opinion either way and we can make it a bit clearer through comments and docs, but I do imagine someone trying to read their own Account Id they just created and not seeing the two most significant bits being what they expected because they are being printed out as little-endian hex strings.